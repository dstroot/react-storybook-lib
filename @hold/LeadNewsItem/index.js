import React from 'react';
import PropTypes from 'prop-types';
import placeholder from './media/logo.png';

const NewsItem = ({ item }) => (
  <>
    <div className="row align-items-center">
      <div className="col d-none d-sm-block">
        <img
          className="img-fluid mr-3"
          src={item.image || placeholder}
          alt="story"
        />
      </div>
      <div className="col">
        <p className="mt-0 text-muted">
          {item.date} / <strong>{item.category}</strong>
        </p>
        <h3 className="news-link font-weight-light">{item.headline}</h3>
        <p>{item.lede}</p>
        <a className="btn btn-primary" href="/news/article">
          Read Article
        </a>
      </div>
    </div>
  </>
);

export default NewsItem;

// define props
NewsItem.propTypes = {
  item: PropTypes.object.isRequired,
};
