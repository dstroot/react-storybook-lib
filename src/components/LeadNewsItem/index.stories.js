import React from "react";
import { storiesOf } from "@storybook/react";

// components
import LeadNewsItem from "../LeadNewsItem";
import image1 from "./media/wildfire.jpg";
const data = {
  data: {
    datetime: "2019-01-08",
    image: image1,
    cutline:
      "Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires",
    headline:
      "Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires",
    category: "Company News"
  }
};

// LeadNewsItem section
storiesOf("LeadNewsItem", module)
  .addParameters({
    info: {
      inline: true,
      header: false
    }
  })
  .add("show LeadNewsItem", () => <LeadNewsItem item={data} />);
