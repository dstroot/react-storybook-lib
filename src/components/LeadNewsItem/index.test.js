import React from 'react';
import renderer from 'react-test-renderer';

import LeadNewsItem from '../LeadNewsItem';
const data = {
  id: '1',
  date: '11-14-18',
  image: '/media/wildfire.jpg',
  url:
    'https://www.pacificlife.com/press-releases/pacific-life-foundation-donates--250-000-to-the-american-red-cro.html',
  category: 'Company News',
  headline:
    'Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires',
  lede:
    'NEWPORT BEACH, CA â€“ The Pacific Life Foundation announced today a $250,000 donation to the American Red Cross in support of relief efforts for those impacted by the California wildfires. The Foundation also announced plans to match employee donations.',
};

describe('LeadNewsItem', () => {
  it('it should render', () => {
    const component = renderer.create(<LeadNewsItem item={data} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
