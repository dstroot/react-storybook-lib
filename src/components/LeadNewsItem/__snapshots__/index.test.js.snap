// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeadNewsItem it should render 1`] = `
<div
  className="row align-items-center"
>
  <div
    className="col d-none d-sm-block"
  >
    <img
      alt="story"
      className="img-fluid mr-3"
      src="/media/wildfire.jpg"
    />
  </div>
  <div
    className="col"
  >
    <p
      className="mt-0 text-muted"
    >
      11-14-18
       / 
      <strong>
        Company News
      </strong>
    </p>
    <h3
      className="news-link font-weight-light"
    >
      Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires
    </h3>
    <p>
      NEWPORT BEACH, CA â€“ The Pacific Life Foundation announced today a $250,000 donation to the American Red Cross in support of relief efforts for those impacted by the California wildfires. The Foundation also announced plans to match employee donations.
    </p>
    <a
      className="btn btn-primary"
      href="/news/article"
    >
      Read Article
    </a>
  </div>
</div>
`;
